app/models/host/discovered.rb

```diff
@@ -84,10 +86,23 @@ def attributes_to_import_from_facts
     super
   end
 
-  def populate_fields_from_facts(facts = self.facts_hash, ignored_type = nil)
-    # detect interfaces and primary interface using extensions
-    parser = super(facts, :foreman_discovery)
+  def populate_fields_from_facts(*params)
+    if SETTINGS[:version].version > '1.16'
```

Comment:
Backward compatibility is great but there might have been more changes that make the next version of discovery compatible only with 1.17 anyway.
I'll leave it up to @lzap to decide if this is needed.